internal/resources/akscluster/schema.go

```diff
@@ -653,11 +728,40 @@ var AutoScaleConfig = &schema.Resource{
 			Type:        schema.TypeInt,
 			Description: "Minimum node count",
 			Optional:    true,
+			ValidateDiagFunc: func(v any, path cty.Path) diag.Diagnostics {
+				return validateIntValueIsPositive(v.(int), "nodepool.auto_scaling_config.min_count must be a positive int32 value")
+			},
 		},
 		maxCountKey: {
 			Type:        schema.TypeInt,
 			Description: "Maximum node count",
 			Optional:    true,
+			ValidateDiagFunc: func(v any, path cty.Path) diag.Diagnostics {
+				return validateIntValueIsPositive(v.(int), "nodepool.auto_scaling_config.max_count must be a positive int32 value")
+			},
 		},
 	},
 }
+
+func validateSelectedValue(allowedValues []string, selectedValue string, selectedKey string) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	for _, name := range allowedValues {
+		if name == selectedValue {
+			return diags
+		}
+	}
+
+	return append(diags, diag.Errorf("%s must be one of: %v", selectedKey, allowedValues)[0])
+}
+
+func validateIntValueIsPositive(value int, errorMessage string) diag.Diagnostics {
```

Comment:
Can you take a look if any of the existing validation methods from terraform-plugin-sdk can be used here: https://github.com/hashicorp/terraform-plugin-sdk/tree/main/helper/validation 

Atleast for int validation you can use it.

Reply:
@ramya-bangera 

Took your request and rewrote the schema validation to use existing validators. Note that ValidateFunc and is deprecated so I had to wrap the built in validators in ToDiagFunc so as to be compatible with ValidateDiagFunc.

Updated the screenshots in the MR commentary to show the new error messages as the text has changed.

Please review and approve.